plugin.audio.radio.roks/radioroks.py

```diff
@@ -0,0 +1,127 @@
+# coding=utf-8
+# Kodi addon for radio Roks online streams
+# developed by E.Kuzin
+
+#  Copyright 2017 Eugene Kuzin
+#
+#  This file is part of the radio.roks Kodi plugin.
+#
+#  This plugin is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This plugin is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this plugin.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import urllib
+import urlparse
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+import xml.etree.ElementTree as ET
+import random
+
+VERSION = '0.1.0'
+
+def build_url(query):
+    base_url = sys.argv[0]
+    return base_url + '?' + urllib.urlencode(query)
+
+
+def play_song(url, icon):
+    # set the path of the song to a list item
+    play_item = xbmcgui.ListItem(path=url)
+    play_item.setArt({'thumb': icon})
+    # the list item is ready to be played by Kodi
+    xbmcplugin.setResolvedUrl(addon_handle, True, listitem=play_item)
+
+
+# build playlist
+def parse_channels():
+    play_list = []
+    tree = ET.parse(addon_path + '/resources/data/channels.xml')
```

Comment:
can you use os.path.join here instead of just concatenating strings? something like `os.path.join(addon_path, "resources","data","channels.xml")` ? Don't forget to import os